src/java.base/share/classes/jdk/internal/util/random/RandomSupport.java

```diff
@@ -715,9 +725,15 @@ public static double boundedNextDouble(RandomGenerator rng, double bound) {
     public static float boundedNextFloat(RandomGenerator rng, float origin, float bound) {
         float r = rng.nextFloat();
         if (origin < bound) {
-            r = r * (bound - origin) + origin;
+            if (bound - origin < Float.POSITIVE_INFINITY) {
+                r = r * (bound - origin) + origin;
+            } else {
+                /* avoids overflow at the cost of 3 more multiplications */
+                float halfOrigin = 0.5f * origin;
+                r = (r * (0.5f * bound - halfOrigin) + halfOrigin) * 2.0f;
```

Comment:
This could be done in double arithmetic, but I think it is better to keep it in float arithmetic for similarity with the code in the double version of the method.